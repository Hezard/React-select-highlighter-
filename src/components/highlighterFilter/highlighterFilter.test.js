import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import HighlighterFilter from './highlighterFilter';

describe('Highlighter Filter', () => {
  let container;
  const props = {
    className: 'filter-test',
    data: [
      {
        id: '0',
        value: 'dummy text',
        variant: 'green',
        start: 22,
        end: 32
      },
      {
        id: '1',
        value: 'type and scrambled',
        variant: 'red',
        start: 194,
        end: 212
      },
      {
        id: '2',
        value: 'unchanged.',
        variant: 'green',
        start: 356,
        end: 366
      }
    ]
  };

  beforeEach(() => {
    ({ container } = render(<HighlighterFilter {...props} />));
  });

  afterEach(() => {
    cleanup();
  });

  test('test initial render', () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});
