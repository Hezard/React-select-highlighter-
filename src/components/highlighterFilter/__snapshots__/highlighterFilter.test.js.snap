// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlighter Filter test initial render 1`] = `
<section
  class="highlighter-filter filter-test"
>
  <div
    class="highlighter__selection highlighter__selection--green"
  >
    dummy text
  </div>
  <div
    class="highlighter__selection highlighter__selection--red"
  >
    type and scrambled
  </div>
  <div
    class="highlighter__selection highlighter__selection--green"
  >
    unchanged.
  </div>
</section>
`;
